src/Agent.Worker/TaskRunner.cs

```diff
@@ -114,6 +115,21 @@ public async Task RunAsync()
                 // Setup container stephost and the right runtime variables for running job inside container.
                 if (stepTarget is ContainerInfo containerTarget)
                 {
+                    if (Stage == JobRunStage.PostJob
+                        && AgentKnobs.SkipPostExeceutionIfTargetContainerStopped.GetValue(ExecutionContext).AsBoolean())
+                    {
+                        // Check that the target container is still running, if not Skip task execution
+                        IDockerCommandManager dockerManager = HostContext.GetService<IDockerCommandManager>();
+                        bool isContainerRunning = await dockerManager.IsContainerRunning(ExecutionContext, containerTarget.ContainerId);
+                        
+                        if (!isContainerRunning)
```

Comment:
Just a suggestion:
Maybe it makes sense to rename the `IsContainerRunning` method to `IsContainerStopped` and also invert returned value. I think it will make the code more readable. What do you think about it?

Reply:
The only thing I don't like here is that when container is not in Running state it could be in a number of other states. If we name method as `IsContainerRunning` and left other logic the same we can exactly say that the container is in running state, but if we invert returned value and change the name to `IsContainerStopped` - several container states will correspond 'true' value here. In my opinion current realization variant is describing result of the method a little bit more clearer.